Mathlib/Tactic/Bound/Attribute.lean

```diff
@@ -0,0 +1,139 @@
+/-
+Copyright (c) 2024 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+import Mathlib.Algebra.Group.ZeroOne
+import Mathlib.Tactic.Bound.Init
+import Qq
+
+/-!
+# The `bound` attribute
+
+Any lemma tagged with `@[bound]` is registered as an apply rule for the `bound` tactic, by
+converting it to either `norm apply` or `safe apply <priority>`.  The classification is based
+on the number and types of the lemma's hypotheses.
+-/
+
+open Lean (MetaM)
+open Qq
+
+namespace Bound
```

Comment:
I think we should put all the tactic code in `Mathlib.Tactic.Bound`

Reply:
I don’t follow this comment. The only code in Attribute.lean is that used for the attributes, which are included early in Mathlib in files which should not include Bound.lean.

Reply:
I'm saying to change this to `namespace Mathlib.Tactic.Bound` and change the other `namespace` in the other file too.

Reply:
Is that the standard? It looks like some tactics don't do it (Linarith) and others do. I'll change it now I suppose.

Reply:
I think we should aim to have all meta code at least in the `Mathlib.Tactic` namespace. I try to put my own tactics in `Mathlib.Tactic.MyTacticName`, but it looks like I've lapsed here and there...

Precise namespacing doesn't hurt, that's for sure!

Reply:
Done!